### PR TITLE
Add dynamic reconfigure

### PR DESCRIPTION
**Pull request by gkouros**
Add dynamic reconfigure for the parameters of the eband planner, trajectory controller and visualizer.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@ find_package(catkin REQUIRED
     roscpp
     tf
     tf_conversions
+    dynamic_reconfigure
```
Comment (jack-oquin) [inline CMakeLists.txt]: Please alphabetize things like this.

File: package.xml [hunk 2]
```diff
@@ -33,6 +33,7 @@
   <build_depend>roscpp</build_depend>
   <build_depend>tf</build_depend>
   <build_depend>tf_conversions</build_depend>
+  <build_depend>dynamic_reconfigure</build_depend>
```
Comment (jack-oquin) [inline package.xml]: Please alphabetize these, too.

**Comment by jack-oquin** [comment]
Thanks for this PR, it looks like a useful enhancement.

At first glance, the code looks fine (with very minor comments). 

But, someone needs to spend enough time reviewing this change to make sure all the parameters have retained their original default values.

**Comment by gKouros** [comment]
Sorry about that, I didn't notice the pattern.

**Comment by jack-oquin** [comment]
It's no big deal, but it's nice to keep lists of dependencies ordered. Makes it easier to see what's there and not there, especially if the lists get long, which this one is not.

**Comment by gKouros** [comment]
I rechecked the initial values and bounds of the parameters and noticed that I accidentally used 0.0 lower bound for the gains k_int and k_diff. However, both parameters as well as the tolerance_timeout parameter are not actually used anywhere in the code. Should I remove all three, or simply fix the lower bounds of the two gain parameters?

**Comment by jack-oquin** [comment]
If you are certain they are not used, they should probably be removed.

**Comment by gKouros** [comment]
I removed the unused parameters. Now, everything looks ok to me, but as you suggested, a second pair of eyes would be advisable.

**Comment by piyushk** [comment]
@jack-oquin: I can take a look and test out this PR.

**Comment by piyushk** [comment]
I've just tested this PR, and it looks great to me. Thanks @gkouros for the contribution!

I'll merge it in, and updated documentation on the ROS wiki page with parameter descriptions.

**Pull request merged**